packages/password-field/src/vaadin-password-field-mixin.js

```diff
@@ -0,0 +1,257 @@
+            return {
+              reveal: 'Show password',
+            };
+          },
+        },
+      };
+    }
+
+    static get observers() {
+      return ['__i18nChanged(i18n.*)'];
+    }
+
+    constructor() {
+      super();
+      this._setType('password');
+      this.__boundRevealButtonClick = this._onRevealButtonClick.bind(this);
+      this.__boundRevealButtonMouseDown = this._onRevealButtonMouseDown.bind(this);
+      this.__lastChange = '';
+    }
+
+    /** @protected */
+    get slotStyles() {
+      const tag = this.localName;
+      return [
+        ...super.slotStyles,
+        `
+          ${tag} [slot="input"]::-ms-reveal {
+            display: none;
+          }
+        `,
+      ];
+    }
+
+    /** @protected */
+    get _revealNode() {
+      return this._revealButtonController && this._revealButtonController.node;
+    }
+
+    /** @protected */
+    ready() {
+      super.ready();
+
+      this._revealPart = this.shadowRoot.querySelector('[part="reveal-button"]');
+
+      this._revealButtonController = new SlotController(this, 'reveal', 'vaadin-password-field-button', {
+        initializer: (btn) => {
+          btn.disabled = this.disabled;
+
+          btn.addEventListener('click', this.__boundRevealButtonClick);
+          btn.addEventListener('mousedown', this.__boundRevealButtonMouseDown);
+        },
+      });
+      this.addController(this._revealButtonController);
+
+      this.__updateAriaLabel(this.i18n);
+
+      this._updateToggleState(false);
+      this._toggleRevealHidden(this.revealButtonHidden);
+
+      if (this.inputElement) {
+        this.inputElement.autocapitalize = 'off';
+      }
+    }
+
+    /**
+     * Override an event listener inherited from `InputControlMixin`
+     * to store the value at the moment of the native `change` event.
+     * @param {Event} event
+     * @protected
+     * @override
+     */
+    _onChange(event) {
+      super._onChange(event);
+
+      this.__lastChange = this.inputElement.value;
+    }
+
+    /**
+     * Override method inherited from `FocusMixin` to mark field as focused
+     * when focus moves to the reveal button using Shift Tab.
+     * @param {Event} event
+     * @return {boolean}
+     * @protected
+     */
+    _shouldSetFocus(event) {
+      return event.target === this.inputElement || event.target === this._revealNode;
+    }
+
+    /**
+     * Override method inherited from `FocusMixin` to not hide password
+     * when focus moves to the reveal button or back to the input.
+     * @param {Event} event
+     * @return {boolean}
+     * @protected
+     */
+    _shouldRemoveFocus(event) {
+      return !(
+        event.relatedTarget === this._revealNode ||
+        (event.relatedTarget === this.inputElement && event.target === this._revealNode)
+      );
+    }
+
+    /**
+     * Override method inherited from `FocusMixin` to toggle password visibility.
+     * @param {boolean} focused
+     * @protected
+     * @override
```

Comment:
**thought:** It can be now unclear where `FocusMixin` and others come from. Previously, it was clear because `PasswordField` extended `TextField`. Now, `PasswordFieldMixin` doesn't extend anything.

Reply:
Updated the mixin and added 3 mixins it depends on (they all use `dedupingMixin` so it should be fine).

Reply:
👍 let's also add `SlotStylesMixin` then for consistency.

Reply:
Done.